.buildkite/pipeline.yml

```diff
@@ -19,10 +27,31 @@ steps:
   - wait: ~
     continue_on_failure: true
 
-  - label: "Junit annotate"
+  - label: ":pipeline: Trigger Integration tests"
+    command: ".buildkite/pipeline.trigger.integration.tests.sh | buildkite-agent pipeline upload"
```

Comment:
There is a bit of complexity here. To run integration tests we have this stack of scripts:
* `.buildkite/pipeline.yml`
* `pipeline.trigger.integration.tests.sh`
* `./.buildkite/scripts/integration_tests.sh`
* `make` targets
* `scripts/test-...`

There would be any option to reduce this complexity? Maybe we can remove the make targets and call directly the scripts? Though the make targets can be useful to reproduce locally with the same parameters.

Or most of the things `integration_tests.sh` does is to install things, could this be also done by providing an image with these dependencies?

Food for thought, no need to do anything about this on this PR.

Reply:
That's true, there are more scripts than before with Jenkins. All that logic was defined inside the same groovy file (`.ci/Jenkinsfile`) i.e. triggering dynamically stages in Jenkins depending on the existing test folders, using specific commands in each one.

> Maybe we can remove the make targets and call directly the scripts? Though the make targets can be useful to reproduce locally with the same parameters.

As you mention, I would keep `make` targets since they are helpful to run and test everything locally.

> Or most of the things integration_tests.sh does is to install things, could this be also done by providing an image with these dependencies?

Those dependencies installed in `integration_tests.sh` probably they could be moved to a custom agent image. It would also make the builds faster since each step would not install everything.